src/Database/FieldTypeConverter.php

```diff
@@ -82,8 +82,8 @@ public function __construct(TypeMap $typeMap, Driver $driver)
             // the interface. Users can implement the TypeInterface instead to have
             // access to this feature.
             $batchingType = $type instanceof BatchCastingInterface &&
-                $type instanceof Type &&
-                strpos(get_class($type), 'Cake\Database\Type') === false;
+                !($type instanceof Type &&
+                strpos(get_class($type), 'Cake\Database\Type') === false);
```

Comment:
Should we use the actual ::class instead of 'Cake\Database\Type' string?

Reply:
no, I just want to check that the class is not inside the cake's namespace. Not checking for any class in particular.